.github/workflows/coverage_tests.yml

```diff
@@ -28,16 +28,35 @@ jobs:
 
       - name: Set up Python
         uses: actions/setup-python@v6
+        id: setup_python
         with:
           python-version: '3.14'
-          cache: 'pip'
-          cache-dependency-path: 'tests/requirements/py3.txt'
+
+      - name: Restore cached virtualenv
+        id: cache-venv
+        uses: actions/cache/restore@v4
+        with:
+          key: venv-coverage-${{ runner.os }}-${{ steps.setup_python.outputs.python-version }}-${{ hashFiles('tests/requirements/py3.txt') }}
+          path: .venv
 
       - name: Install dependencies
         run: |
+          if (-not (Test-Path .venv)) {
+            python -m venv .venv
+          }
```

Comment:
Can we replace this with the same cache-hit condition? This avoids using windows powershell here. (We are likely to build out a matrix across other OSes.)